tests/test_wasting.py

```diff
@@ -0,0 +1,827 @@
+    """Check: Wasting onset --> death"""
+    """ Check if the risk of death is 100% does everyone with SAM die? """
+    dur = pd.DateOffset(days=0)
+    popsize = 1000
+    sim = get_sim(tmpdir)
+
+    sim.make_initial_population(n=popsize)
+    sim.simulate(end_date=start_date + dur)
+    sim.event_queue.queue = []  # clear the queue
+
+    # get wasting module
+    wasting_module = sim.modules['Wasting']
+    # Make 100% death rate by replacing with empty linear model 1.0
+    wasting_module.sam_death_equation = LinearModel(
+        LinearModelType.MULTIPLICATIVE, 1.0)
+
+    # Get the children to use:
+    df = sim.population.props
+    under5s = df.loc[df.is_alive & (df['age_years'] < 5)]
+    person_id = under5s.index[0]
+
+    # re-set parameter values to make zero recovery rate and 100% death rate
+    wasting_module.parameters['recovery_rate_with_standard_RUTF'] = 0.0
+    wasting_module.parameters['recovery_rate_with_inpatient_care'] = 0.0
+    wasting_module.parameters['base_death_rate_untreated_SAM'] = 1.0
+
+    # re-initialise wasting models
+    wasting_module.pre_initialise_population()
+
+    # make an individual diagnosed as SAM by WHZ category.
+    # We want to make this individual qualify for death
+    df.loc[person_id, 'un_ever_wasted'] = True
+    df.loc[person_id, 'un_WHZ_category'] = 'WHZ<-3'
+    df.loc[person_id, 'un_clinical_acute_malnutrition'] = 'SAM'
+
+    # apply wasting symptoms to this individual
+    wasting_module.wasting_clinical_symptoms(person_id)
+
+    # check symptoms are applied
+    assert person_id in set(
+        sim.modules['SymptomManager'].who_has('weight_loss'))
+
+    # run health seeking behavior and ensure non-emergency event is scheduled
+    hsp = HealthSeekingBehaviourPoll(sim.modules['HealthSeekingBehaviour'])
+    hsp.run()
+
+    # check non-emergency care event is scheduled
+    assert isinstance(
+        sim.modules['HealthSystem'].find_events_for_person(person_id)[0][1],
+        hsi_generic_first_appts.HSI_GenericNonEmergencyFirstAppt)
+
+    # Run the created instance of HSI_GenericFirstApptAtFacilityLevel0
+    # and check care was sought
+    ge = [ev[1] for ev in
+          sim.modules['HealthSystem'].find_events_for_person(person_id) if
+          isinstance(ev[1],
+                     hsi_generic_first_appts.HSI_GenericNonEmergencyFirstAppt)
+          ][0]
+    ge.run(squeeze_factor=0.0)
+
+    # check inpatient care event is scheduled
+    assert isinstance(
+        sim.modules['HealthSystem'].find_events_for_person(person_id)[0][1],
+        HSI_Wasting_OutpatientTherapeuticProgramme_SAM)
+
+    # Run the created instance of
+    # HSI_Wasting_OutpatientTherapeuticProgramme_SAM and check care was sought
+    sam_ev = [ev[1] for ev in
+              sim.modules['HealthSystem'].find_events_for_person(person_id) if
+              isinstance(ev[1],
+                         HSI_Wasting_OutpatientTherapeuticProgramme_SAM)][0]
+    sam_ev.run(squeeze_factor=0.0)
```

Comment:
@tbhallett. since we have removed natural death, I think we need that. Currently, death is as a result of treatment failure hence it is involving HSI events.